keps/sig-cli/2985-public-krm-functions-registry/README.md

```diff
@@ -0,0 +1,996 @@
+popular in the k8s configuration management space.
+
+Google has a [GitHub repository](https://github.com/GoogleContainerTools/kpt-functions-catalog)
+under GoogleContainerTools for hosting the source of the functions and a
+[website](https://catalog.kpt.dev/) for presenting the functions to the end
+users. However, not everyone in the community can contribute to it for various
+reasons. E.g. a company’s policy may not allow its employees to contribute to
+repo owned by another company.
+
+To have a thriving ecosystem of KRM functions, we must enable contributions for
+all community members in vendor-neutral registry. With a public KRM function
+registry, we can significantly improve the Day 0 and Day 2 user experiences. On
+Day 0, users can discover some KRM functions that may be useful for their needs.
+On Day 2, users can query the registry to discover new function versions.
+
+[KRM functions]: https://github.com/kubernetes-sigs/kustomize/blob/master/cmd/config/docs/api-conventions/functions-spec.md
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+* Enable end users of orchestrators (e.g. Kustomize, Kpt) that support KRM
+  functions to discover and leverage a common ecosystem of compatible functions.
+* Enable end users to discover and use sets of functions specifically from 
+  publishers they trust.
+* Enable function authors from any company to expose their function in a
+  well-known index for discovery by end users.
+* Provide a central place for first-party (SIG-sponsored) plugins to be built
+  and added to the index.
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+* Replace or compete with Catalog as publication format for collections of
+  functions. Catalog should be used by this Registry, and the details of its
+  internal format should be discussed in that KEP.
+* Support building non-SIG-sponsored functions.
+* Support SIG-sponsored functions written in a language other than Go or
+  published in a format other than containerized.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+This KEP proposes to create a public KRM functions registry. There will be 2
+components to back the registry:
+
+- A repo sponsored by sig-cli to host the index of the functions in the registry.
+- A website that presents the doc and examples to the end users and allows users
+  to search and discover KRM functions.
+
+
+### Publisher
+
+A publisher can be a project/community in Kubernetes or a company. For example,
```

Comment:
I'm still unclear on the line elsewhere about functions without a publisher. Why can't individuals be publishers? Might help to describe how publishers will be defined and tracked by the registry. You mention company email addresses as one aspect of this. Wdyt about involving Github orgs as an option?

Reply:
> Why can't individuals be publishers?

I guess we don't want to prevent that. But IMO it may not be very useful. Do you have an use case in mind?

> Wdyt about involving Github orgs as an option?

I like this idea. We can leverage that. A user must be a member of the GitHub org to publish functions for the org.

Reply:
👍  to the GitHub org idea